docs/site/src/plugins/inject-code/injectLoader.js

```diff
@@ -0,0 +1,204 @@
+    const matches = fileContent.match(/\{@\w+: .+\}/g);
+    if (matches) {
+      matches.forEach((match) => {
+        const replacer = new RegExp(match, "g");
+        if (match.startsWith("{@inject: ")) {
+          const injectFileFull = match.substring(10, match.length - 1);
+          const injectFile = injectFileFull.substring(
+            0,
+            injectFileFull.indexOf("#") > 0
+              ? injectFileFull.indexOf("#")
+              : injectFileFull.length,
+          );
+          let fileExt = injectFile.substring(injectFile.lastIndexOf(".") + 1);
+          let language = "";
+          const fullPath = path.join(repoPath, injectFile);
+
+          // Assuming rust manifests and locks never being used
+          switch (fileExt) {
+            case "toml":
+              language = "move";
+              break;
+            case "lock":
+              language = "move";
+              break;
+            case "sh":
+              language = "shell";
+              break;
+            case "mdx":
+              language = "markdown";
+              break;
+            case "tsx":
+              language = "ts";
+              break;
+            default:
+              language = fileExt;
+          }
+
+          if (fs.existsSync(fullPath)) {
+            let injectFileContent = fs.readFileSync(fullPath, "utf8");
+            const marker =
+              injectFileFull.indexOf("#") > 0
+                ? injectFileFull.substring(injectFileFull.indexOf("#"))
+                : null;
+            if (marker) {
+              const regexStr = `\\/\\/\\s?docs::${marker.trim()}\\b([\\s\\S]*)\\/\\/\\s*docs::\\/\\s?${marker.trim()}\\b`;
+              const closingsStr = `\\/\\/\\s?docs::\\/${marker.trim()}\\b([)};]*)`;
+
+              const closingRE = new RegExp(closingsStr, "g");
+
+              const regex = new RegExp(regexStr, "g");
+              const match = regex.exec(injectFileContent);
+              const closingStr = closingRE.exec(injectFileContent);
+              if (match) {
+                injectFileContent = match[1];
+              }
+              if (closingStr) {
+                const closingTotal = closingStr[1].length;
+                let closingArray = [];
+                for (let i = 0; i < closingTotal; i++) {
+                  const currentChar = closingStr[1][i];
+                  const nextChar = closingStr[1][i + 1];
+
+                  if (nextChar === ";") {
+                    closingArray.push(currentChar + nextChar);
+                    i++;
+                  } else {
+                    closingArray.push(currentChar);
+                  }
+                }
+                const totClosings = closingArray.length;
+
+                // Process any closing elements added in the closing comment of source code
+                let closing = "";
+                for (let j = 0; j < totClosings; j++) {
+                  let space = "  ".repeat(totClosings - 1 - j);
```

Comment:
This assumes a two-space indent, but note that we tend to use 4 spaces in Move and Rust, and tabs in TypeScript.